mne/io/base.py

```diff
@@ -2058,3 +2059,119 @@ def _check_update_montage(info, montage):
+        MISC channels. Values should correspond to channels listed within
+        the raw file.
+    reference : bool | str
+        If True, add average EEG reference projector (if it's not already
+        present).
+    preload : bool or str (default False)
+        Preload data into memory for data manipulation and faster indexing.
+        If True, the data will be preloaded into memory (fast, requires
+        large amount of memory). If preload is a string, preload is the
+        file name of a memory-mapped file which is used to store the data
+        on the hard drive (slower, requires less memory).
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+    **kwargs :
+        Keyword arguments to pass to read_raw_* (see system-specific reader
+        for more details).
+
+    Returns
+    -------
+    raw : Instance of Raw
+        A Raw object.
+    """
+
+    from mne.io import (read_raw_brainvision, read_raw_bti, read_raw_edf,
+                        read_raw_egi, read_raw_fif, read_raw_kit)
+
+    if isinstance(input_fname, string_types):
+        fname, ext = os.path.splitext(input_fname)
+    elif isinstance(input_fname, list):
+        for f in input_fname:
+            fname, ext = os.path.splitext(input_fname)
+            err = "File must be '.fif', not %s." % ext
+            assert_equal(ext, '.fif', err)
+
+    if eog is not None:
+        eog_map = dict()
+        for s in eog:
+            eog_map[s] = 'eog'
+    if misc is not None:
+        misc_map = dict()
+        for s in misc:
+            misc_map[s] = 'misc'
+
+    # (MEG) KIT systems
+    if ext in ['.con', '.sqd']:
+        raw = read_raw_kit(input_fname, preload=preload, verbose=verbose,
+                           **kwargs)
+        if eog is not None:
+            raw.set_channels_type(eog_map)
+        if misc is not None:
+            raw.set_channels_type(misc_map)
+    # (EEG) EDF and Biosemi systems
+    elif ext in ['bdf', '.edf']:
+        raw = read_raw_edf(input_fname, montage=montage, eog=eog, misc=eog,
+                  preload=preload, verbose=verbose, **kwargs)
+    # (EEG) EGI systems
+    elif ext == '.raw':
+        raw = read_raw_egi(input_fname, montage=montage, eog=eog, misc=misc,
+                  preload=preload, verbose=verbose, **kwargs)
+    # (MEG) Neuromag or converted-to-fif systems
+    elif ext in ['.fif', '.gz']:
+        raw = read_raw_fif(input_fname, add_eeg_ref=reference,
+                  preload=preload, verbose=verbose, **kwargs)
+        if eog is not None:
+            raw.set_channels_type(eog_map)
+        if misc is not None:
+            raw.set_channels_type(misc_map)
+    # (MEG) BTi systems
+    elif ext == '':
+        if os.path.isfile(input_fname):
+            raw = read_raw_bti(input_fname, preload=preload, verbose=verbose,
+                               **kwargs)
+        else:
+            raise IOError('This file does not exist.')
```

Comment:
This can be simplified (save one whole line!) by doing:
```
if not op.isfile(...):
    error
raw = read_raw
```
It's a nice pattern to get into because in other contexts it can save you a lot of unnecessary indentation levels.